confignode/src/main/java/org/apache/iotdb/confignode/persistence/node/NodeStatistics.java

```diff
@@ -0,0 +1,121 @@
+ */
+package org.apache.iotdb.confignode.persistence.node;
+
+import org.apache.iotdb.commons.cluster.NodeStatus;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+public class NodeStatistics {
+
+  // For guiding queries, the higher the score the higher the load
+  private long loadScore;
+
+  // The current status of the Node
+  private NodeStatus status;
+  // The reason why lead to the current NodeStatus (for showing cluster)
+  // Notice: Default is null
+  private String statusReason;
+
+  public NodeStatistics() {
+    // Empty constructor
+  }
+
+  public NodeStatistics(long loadScore, NodeStatus status, String statusReason) {
+    this.loadScore = loadScore;
+    this.status = status;
+    this.statusReason = statusReason;
+  }
+
+  public long getLoadScore() {
+    return loadScore;
+  }
+
+  public NodeStatus getStatus() {
+    return status;
+  }
+
+  public String getStatusReason() {
+    return statusReason;
+  }
+
+  public void setRemoving() {
+    this.statusReason = null;
+    this.status = NodeStatus.Removing;
+  }
+
+  public void serialize(OutputStream stream) throws IOException {
+    ReadWriteIOUtils.write(loadScore, stream);
+    ReadWriteIOUtils.write(status.getStatus(), stream);
+    if (statusReason != null) {
+      ReadWriteIOUtils.write(true, stream);
+      ReadWriteIOUtils.write(statusReason, stream);
+    } else {
+      ReadWriteIOUtils.write(false, stream);
+    }
+  }
+
+  // Deserializer for consensus-write
+  public void deserialize(ByteBuffer buffer) {
+    loadScore = buffer.getLong();
+    status = NodeStatus.parse(ReadWriteIOUtils.readString(buffer));
+    if (ReadWriteIOUtils.readBool(buffer)) {
+      statusReason = ReadWriteIOUtils.readString(buffer);
+    } else {
+      statusReason = null;
+    }
+  }
+
+  public void deserialize(InputStream inputStream) throws IOException {
+    loadScore = ReadWriteIOUtils.readLong(inputStream);
+    status = NodeStatus.parse(ReadWriteIOUtils.readString(inputStream));
+    if (ReadWriteIOUtils.readBool(inputStream)) {
+      statusReason = ReadWriteIOUtils.readString(inputStream);
+    } else {
+      statusReason = null;
+    }
+  }
+
+  public static NodeStatistics generateDefaultNodeStatistics() {
+    return new NodeStatistics(Long.MAX_VALUE, NodeStatus.Unknown, null);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    NodeStatistics that = (NodeStatistics) o;
+    return status == that.status
+        && Objects.equals(statusReason, that.statusReason)
+        // In order to prevent the NodeStatistics from updating too fast.
+        // Here we consider the loadScore equal when the difference is small.
+        // TODO: optimize
+        && Math.abs(loadScore - that.loadScore) < 60 * 1000;
```

Comment:
I agree with that. However, because we judge the equality of NodeStatistics and RegionGroupStatistics to decide whether to update the LoadStatistics. The LoadStatistics will be updated too fast if we don't check the `loadScore` in NodeStatistics and `versionTimestamp` in RegionStatus like that.  Therefore, I currently use this method and left a TODO here.